### ISSUE TITLE
Sampling bug when one of the choices in `categorical` has probability 1

### ISSUE THREAD

**Issue by ztangent** [probcomp/Gen.jl#256]
There seems to be a bug with sampling from categorical when the probabilities given place all mass on a single distribution. It looks like this caused by a buggy upstream implementation of sampling from a categorical in `Distributions.jl` :
```julia
julia> sum([categorical([1, 0]) for i in 1:100])
155
julia> sum([categorical([0.9999, 0.0001]) for i in 1:100])
100
julia> sum([rand(Gen.Distributions.Categorical([1, 0])) for i in 1:100])
155
```
Will open a corresponding issue in `Distributions.jl` if this bug is present in the latest version.

**Comment by ztangent** [comment]
Yup, just confirmed that the bug is also present in the latest version of Distributions.jl: https://github.com/JuliaStats/Distributions.jl/issues/1111

**Comment by ztangent** [comment]
As a hot-fix, it seems like passing in 1.0 instead of 1 leads to the correct behavior:
```
julia> sum([categorical([1.0, 0]) for i in 1:100])
100
```
Thanks @alex-lew for noticing this!

**Comment by jamesonquinn** [comment]
Works for me with Distributions v0.24.10. I'll add your example as a test case to close this bug.
